java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java

```diff
@@ -1970,7 +1976,7 @@ CompletableFuture<List<Diagnostic>> computeDiagnostics(String uri, EnumSet<Error
     private List<Diagnostic> computeDiags(String uri, int offset, ErrorProvider.Kind errorKind, long orgV) {
         List<Diagnostic> result = new ArrayList<>();
         FileObject file = fromURI(uri);
-        if (file == null) {
+        if (file == null || !this.hintsPrefsStatus) {
```

Comment:
So, my understanding is that this will prevent diagnostic computation until the hints setting are read. Which is fine, I think, but:
- this should have a separate comment (either separate it into a separate if, or document each of the conditions)
- the diagnostic computation must be unblocked even if the setting does not exist - please see above.